### DISCUSSION TITLE
pathRewrite is case sensitive. Is there a way to specify to ignoreCase?

### DISCUSSION THREAD

**Discussion by ubreddy** [chimurai/http-proxy-middleware#536] [Q&A]
### Is this a question?

pathRewrite is case sensitive. Is there a way to specify to ignoreCase?

### Is this a bug report?
 No

### Is this a feature request?

No

### Steps to reproduce

currently if we specify the pathRewrite object with key paths are '^/api/v1/CaseInSenSITIVEFragement' : '/api'

the proxy matches only if case sensitivity exactly matches... how to tell the proxy to ignore case...

**Comment by chimurai** [comment]
Instead providing an `object` you can provide a pathRewrite `function` in which can implement the behaviour you want.

Example:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathRewrite.md#custom-rewrite-function

**Comment by ubreddy** [comment]
yes, but that rewrite function is only invoked on the path regex match... 
if I need to have different functions for different paths, and the paths are case insensitive it becomes quite difficult.

e.g. 
'^/api/v1' : fn1
'^/api/v2'  : fn2  etc...
now if I need the paths to be case insensitive and also need different functions to be invoked... it does not work...
the only workaround is
'/.*' : universalFn    

universalFn that has the knowledge of all paths and handles all of them.. it centralizes the logic than having a distributed way of localizing the login...

**Comment by peterjhart** [comment]
So you actually want case-insensitive keys. That smells like you are addressing a problem in the wrong place.
